src/notation/view/partlistmodel.cpp

```diff
@@ -368,28 +300,28 @@ QList<int> PartListModel::selectedRows() const
     return result;
 }
 
-void PartListModel::apply()
+void PartListModel::openSelectedParts()
 {
-    ExcerptNotationList newExcerpts;
-
-    for (INotationPtr notation : m_notations) {
-        IExcerptNotationPtr excerpt = std::dynamic_pointer_cast<IExcerptNotation>(notation);
+    QList<int> rows = selectedRows();
+    if (rows.empty()) {
+        return;
+    }
 
-        if (excerpt) {
-            newExcerpts.push_back(excerpt);
-        }
+    ExcerptNotationList newExcerpts;
+    for (int index : rows) {
+        newExcerpts.push_back(m_excerpts[index]);
     }
 
-    masterNotation()->setExcerpts(newExcerpts);
+    masterNotation()->notation()->undoStack()->prepareChanges();
+    masterNotation()->initExcerpts(newExcerpts);
```

Comment:
Better to move prepareChanges() and commitChanges() to the initExcerpts implementation